go/vt/discovery/healthcheck.go

```diff
@@ -65,28 +66,36 @@ var (
 	healthcheckOnce           sync.Once
 
 	// TabletURLTemplateString is a flag to generate URLs for the tablets that vtgate discovers.
-	TabletURLTemplateString = flag.String("tablet_url_template", "http://{{.GetTabletHostPort}}", "format string describing debug tablet url formatting. See the Go code for getTabletDebugURL() how to customize this.")
+	TabletURLTemplateString string
 	tabletURLTemplate       *template.Template
 
-	// AllowedTabletTypes is the list of allowed tablet types. e.g. {PRIMARY, REPLICA}
+	// AllowedTabletTypes is the list of allowed tablet types. e.g. {PRIMARY, REPLICA}.
 	AllowedTabletTypes []topodata.TabletType
+
 	// KeyspacesToWatch - if provided this specifies which keyspaces should be
 	// visible to the healthcheck. By default the healthcheck will watch all keyspaces.
-	KeyspacesToWatch flagutil.StringListValue
+	KeyspacesToWatch []string
+
+	// tabletFilters are the keyspace|shard or keyrange filters to apply to the full set of tablets.
+	tabletFilters []string
 
-	// tabletFilters are the keyspace|shard or keyrange filters to apply to the full set of tablets
-	tabletFilters flagutil.StringListValue
-	// refreshInterval is the interval at which healthcheck refreshes its list of tablets from topo
-	refreshInterval = flag.Duration("tablet_refresh_interval", 1*time.Minute, "tablet refresh interval")
-	// refreshKnownTablets tells us whether to process all tablets or only new tablets
-	refreshKnownTablets = flag.Bool("tablet_refresh_known_tablets", true, "tablet refresh reloads the tablet address/port map from topo in case it changes")
-	// topoReadConcurrency tells us how many topo reads are allowed in parallel
-	topoReadConcurrency = flag.Int("topo_read_concurrency", 32, "concurrent topo reads")
+	// refreshInterval is the interval at which healthcheck refreshes its list of tablets from topo.
+	// We need to initialize this for tests to pass.
+	refreshInterval = time.Minute
```

Comment:
Notice that I duplicate the default value here. This is because tests depend on this.
What is the recommendation? Should we leave this uninitialized and set it in the tests?

Reply:
In https://github.com/vitessio/vitess/commit/52b74fac6241f95ed3841505a1b861e42f2e9b89 I went with fixing the tests (it's cleaner imo but I could also go either way)

Reply:
I ended up changing the tests.